### PR TITLE
lima: init at 0.4.0

### PR DESCRIPTION
**Pull request by voanhduy1512**
###### Motivation for this change

Add lima

Info from their GitHub page:

```
Lima launches Linux virtual machines with automatic file sharing, port forwarding, and containerd.
Lima can be considered as a some sort of unofficial "macOS subsystem for Linux", or "containerd for Mac".
Lima is expected to be used on macOS hosts, but can be used on Linux hosts as well.
```

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

I would like to have some feedback on where to put the package. Right now I put it under virtualization but not sure that's the correct place. 
And based on https://github.com/AkihiroSuda/lima/blob/434e230631ea534041834e7d6e92511765d0423e/Makefile#L23, Lima builds both amd64 and arm64 version of guestagent, so I need to run the full command in postBuild, not sure if there are any better way to do it as well.

### CODE REVIEW

**Comment by zowoq** [comment]
Using the makefile is fine.
```nix
{ lib, fetchFromGitHub, buildGoModule, installShellFiles, makeWrapper, qemu_kvm }:

buildGoModule rec {
  pname = "lima";
  version = "0.4.0";

  src = fetchFromGitHub {
    owner = "AkihiroSuda";
    repo = pname;
    rev = "v${version}";
    sha256 = "sha256-vwAxVBy2SqghxJGscSZ1o1B8EMvQh1fz5CS1YG7Rq2g=";
  };

  vendorSha256 = "sha256-xM9LLh5c5QBrcIptdqiNNp1nU9GcdQvwrCnnyuXWYfE=";

  nativeBuildInputs = [ installShellFiles makeWrapper ];

  buildPhase = ''
    runHook preBuild
    make "VERSION=v${version}" binaries
    runHook postBuild
  '';

  installPhase = ''
    runHook preInstall
    mkdir -p $out
    cp -r _output/* $out
    wrapProgram $out/bin/limactl \
      --prefix PATH : ${lib.makeBinPath [ qemu_kvm ]}
    installShellCompletion --cmd limactl \
      --bash <($out/bin/limactl completion bash)
    # ensure limactl is in $PATH
    for i in lima nerdctl.lima; do
      wrapProgram $out/bin/$i \
        --prefix PATH : $out/bin
    done
    runHook postInstall
  '';

  meta = with lib; {
    description = "Linux virtual machines";
    homepage = "https://github.com/AkihiroSuda/lima";
    license = licenses.asl20;
  };
}
```

**Comment by voanhduy1512** [comment]
thanks @zowoq for the suggestion. Using project Makefile is definitely better then my approach.

File: pkgs/applications/virtualization/lima/default.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  installPhaseCheck = ''
+    $out/bin/limactl prune 2>&1
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/AkihiroSuda/lima";
+    description = "Linux virtual machines (on macOS, in most cases)";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ anhduy ];
+    platforms = with platforms; linux ++ darwin;
```
Comment (zowoq) [inline pkgs/applications/virtualization/lima/default.nix:50]: Not needed, it's the same as the default set by `buildGoModule`. 
```suggestion
    platforms = with platforms; linux ++ darwin;
```

Comment (voanhduy1512) [reply pkgs/applications/virtualization/lima/default.nix]: got it, thanks

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -690,6 +690,12 @@
       fingerprint = "AA4B 8EC3 F971 D350 482E  4E20 0299 AFF9 ECBB 5169";
     }];
   };
+  anhduy = {
```
Comment (zowoq) [inline maintainers/maintainer-list.nix:693]: Please move this to a separate commit.
```
maintainers: add anhduy
lima: init at 0.4.0
```

Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:693]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

Comment (voanhduy1512) [reply maintainers/maintainer-list.nix:693]: done

Comment (voanhduy1512) [reply maintainers/maintainer-list.nix:693]: done

File: pkgs/applications/virtualization/lima/default.nix [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    mkdir -p $out
+    cp -r _output/* $out
+    wrapProgram $out/bin/limactl \
+      --prefix PATH : ${lib.makeBinPath [ qemu ]}
+    installShellCompletion --cmd limactl \
+      --bash <($out/bin/limactl completion bash)
+    runHook postInstall
+  '';
+
+  installPhaseCheck = ''
```
Comment (zowoq) [inline pkgs/applications/virtualization/lima/default.nix:41]:
```suggestion
  doInstallCheck = true;
  installCheckPhase = ''
```

File: pkgs/applications/virtualization/lima/default.nix [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    $out/bin/limactl prune 2>&1
```
Comment (zowoq) [inline pkgs/applications/virtualization/lima/default.nix:42]: Should really matter as we set $HOME during the build but using a non destructive test is probably safer.

```suggestion
    $out/bin/limactl validate $out/share/doc/lima/examples/default.yaml
```

Comment (voanhduy1512) [reply pkgs/applications/virtualization/lima/default.nix]: I try build without USER and didn't see the error like yours.

Comment (voanhduy1512) [reply pkgs/applications/virtualization/lima/default.nix]: btw, i used that check because that's what they use in their homebrew package

Comment (zowoq) [reply pkgs/applications/virtualization/lima/default.nix]: Same error with `ofborg` https://github.com/NixOS/nixpkgs/pull/128052/checks?check_run_id=2913381618

Comment (voanhduy1512) [reply pkgs/applications/virtualization/lima/default.nix]: You're right, just now I tested on nixos and not macos. Building on macos gives the same error
